test/modules/billing/services/batch-service.js

```diff
@@ -1434,4 +1435,49 @@ experiment('modules/billing/services/batch-service', () => {
       });
     });
   });
+
+  experiment('.getAllCmTransactionsForBatch', () => {
+    const getTransaction = page => ({
+      transactions: [{
+        id: `test-cm-transaction-${page}`
+      }]
+    });
+    let result;
+    beforeEach(async () => {
+      chargeModuleBillRunConnector.getTransactions.onFirstCall().resolves({
+        pagination: { pageCount: 3 },
+        data: getTransaction(1)
+      });
+      chargeModuleBillRunConnector.getTransactions.onSecondCall().resolves({
+        data: getTransaction(2)
+      });
+      chargeModuleBillRunConnector.getTransactions.onThirdCall().resolves({
+        data: getTransaction(3)
+      });
+
+      result = await batchService.getAllCmTransactionsForBatch({ externalId: 'test-cm-batch-id' });
+    });
+
```

Comment:
Might be good to have a little test here to check that:
```
chargeModuleBillRunConnector.getTransactions.callCount
```
has been called the expected number of times based on the number of pages expected

Reply:
Thank you :) I wanted to do a test like this, but must've got distracted and forgot about it in the end! I've added it in now